error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/AssertJThrowingCallableRulesTestInput.java

```diff
@@ -218,4 +219,12 @@ void testAssertThatThrownByIsInstanceOf() {
             .isInstanceOf(IllegalArgumentException.class)
             .withFailMessage(String.format("foo %s %s", "bar", 1)));
   }
+
+  @SuppressWarnings("deprecation" /* Migrates away from deprecated API. */)
+  AbstractThrowableAssert<?, ? extends Throwable> testAbstractThrowableAssertCauseIsSameAs() {
+    IllegalArgumentException illegalArgumentException = new IllegalArgumentException();
+    IllegalStateException illegalStateException =
+        new IllegalStateException(illegalArgumentException);
+    return assertThat(illegalStateException).hasCauseReference(illegalArgumentException);
```

Comment:
Usually we keep these tests minimal, even if nonsensical; will propose something.